licensescout-core/src/main/java/org/aposin/licensescout/license/LicenseCheckedList.java

```diff
@@ -209,26 +209,32 @@ private LicenseResult fetchLicenseResult(final LicenseStoreData licenseStoreData
                                              final String[] values, final String documentationUrl, final Notice notice,
                                              final Provider provider) {
         final int numLicenseNames = values.length - LICENSE_NAMES_OFFSET;
+        LicenseProcessingMode licenseProcessingMode;
         final List<License> licenses = new ArrayList<>();
         if (numLicenseNames == 0) {
-            licenses.add(licenseStoreData.getLicenseBySpdxIdentifier(LicenseSpdxIdentifier.NO_MANUAL_INFORMATION));
+            licenseProcessingMode = LicenseProcessingMode.OVERWRITE_EMPTY_LIST;
         } else {
-            for (int i = 0; i < numLicenseNames; i++) {
-                final String licenseName = values[i + LICENSE_NAMES_OFFSET].trim();
-                final License license = licenseStoreData.getLicenseBySpdxIdentifier(licenseName);
-                if (license == null) {
-                    log.error("License not found in store: '" + licenseName + "'");
-                } else {
-                    licenses.add(license);
+            if (values[LICENSE_NAMES_OFFSET].trim().equals("-")) {
```

Comment:
Yes, because its the result of a split() operation. And before, in line 214, with 'if (numLicenseNames == 0) {' (where numLicenseNames is 'values.length - LICENSE_NAMES_OFFSET') it is verified that the index exists. However, I was chasing and fixing another bug, and in the course of this I did additional restructuring. So please re-review.